src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -219,52 +212,50 @@ protected override void Dispose(bool disposing)
             base.Dispose(disposing);
         }
 
-        private static ActivityDataRequest ComputeActivityDataRequest(
+        private static ActivitySamplingResult ComputeActivitySamplingResult(
             in ActivityCreationOptions<ActivityContext> options,
             Sampler sampler)
         {
-            // As we set ActivityListener.AutoGenerateRootContextTraceId = true,
-            // Parent.TraceId will always be the TraceId of the to-be-created Activity,
-            // if it get created.
-            ActivityTraceId traceId = options.Parent.TraceId;
-
             var samplingParameters = new SamplingParameters(
                 options.Parent,
-                traceId,
+                options.TraceId,
                 options.Name,
                 options.Kind,
                 options.Tags,
                 options.Links);
 
             var shouldSample = sampler.ShouldSample(samplingParameters);
 
-            var activityDataRequest = shouldSample.Decision switch
+            var activitySamplingResult = shouldSample.Decision switch

```

Comment:
would this callback be called for legacy (http-in/out) DiagnosticSource instrumentations?

Reply:
Good point!
I missed it. Let me add it to the other refactor on ActivitySourceAdapter (which deals with Activity created using DiagnosticSource methods)
Tracking here https://github.com/open-telemetry/opentelemetry-dotnet/issues/1210